test/oai/test_mistral.py

```diff
@@ -0,0 +1,261 @@
+
+    # Define the mock response directly within the patch
+    mock_function = MagicMock(name="currency_calculator")
+    mock_function.name = "currency_calculator"
+    mock_function.arguments = '{"base_currency": "EUR", "quote_currency": "USD", "base_amount": 123.45}'
+
+    # Define the mock response directly within the patch
+    mock_create.return_value = MagicMock(
+        choices=[
+            MagicMock(
+                finish_reason="tool_calls",
+                message=MagicMock(
+                    content="",  # Message is empty for tool responses
+                    tool_calls=[MagicMock(id="gdRdrvnHh", function=mock_function)],
+                ),
+            )
+        ],
+        id="mock_mistral_response_id",
+        model="mistral-small-latest",
+        usage=MagicMock(prompt_tokens=10, completion_tokens=20),
+    )
+
+    # Test parameters
+    converted_functions = [
+        {
+            "type": "function",
+            "function": {
+                "description": "Currency exchange calculator.",
+                "name": "currency_calculator",
+                "parameters": {
+                    "type": "object",
+                    "properties": {
+                        "base_amount": {"type": "number", "description": "Amount of currency in base_currency"},
+                        "base_currency": {
+                            "enum": ["USD", "EUR"],
+                            "type": "string",
+                            "default": "USD",
+                            "description": "Base currency",
+                        },
+                        "quote_currency": {
+                            "enum": ["USD", "EUR"],
+                            "type": "string",
+                            "default": "EUR",
+                            "description": "Quote currency",
+                        },
+                    },
+                    "required": ["base_amount"],
+                },
+            },
+        }
+    ]
+
+    mistral_messages = [
+        ChatMessage(
+            role="system",
+            content='For currency exchange tasks,\n        only use the functions you have been provided with.\n        Output \'TERMINATE\' when an answer has been provided.\n        Do not include the function name or result in the JSON.\n        Example of the return JSON is:\n        {\n            "parameter_1_name": 100.00,\n            "parameter_2_name": "ABC",\n            "parameter_3_name": "DEF",\n        }.\n        Another example of the return JSON is:\n        {\n            "parameter_1_name": "GHI",\n            "parameter_2_name": "ABC",\n            "parameter_3_name": "DEF",\n            "parameter_4_name": 123.00,\n        }. ',
+            name=None,
+            tool_calls=None,
+            tool_call_id=None,
+        ),
+        ChatMessage(
+            role="user", content="How much is 123.45 EUR in USD?", name=None, tool_calls=None, tool_call_id=None
+        ),
+    ]
```

Comment:
If this is mock, we should make it as simple as possible.
```suggestion
    mistral_messages = [
        ChatMessage( role="user", content="How much is 123.45 EUR in USD?")
    ]
```

Reply:
Thanks, removing